cursor property is not applying on <area> element

AlexsanderDamaceno opened the pull request:
#### 3445c01a99afadb9688b2871680f308e3eb27012
<pre>
cursor property is not applying on &lt;area&gt; element
<a href="https://bugs.webkit.org/show_bug.cgi?id=221762">https://bugs.webkit.org/show_bug.cgi?id=221762</a>

Reviewed by Ryosuke Niwa.

Match with other browsers behavior for area element.

The `&lt;area&gt;` element has `display: none` by default, so no renderer is set for it.
As a result, the cursor could not be looked up in the style object,
since we access the renderer to retrieve cursor information on each mouse event.
This is resolved by accessing the style information directly from the node object when dealing with `&lt;area&gt;` elements.

* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::selectCursor):

Canonical link: <a href="https://commits.webkit.org/290169@main">https://commits.webkit.org/290169@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fed97b8b4b3af96b1c75e53a08e305ff1ec54abe

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/88492 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8011 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/42932 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/93450 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/39246 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/90543 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/8398 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/16195 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68252 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25972 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/91494 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6427 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/80043 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/48619 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6199 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/34459 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/38354 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/76572 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/35342 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/95292 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/15667 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/11708 "Found 1 new test failure: http/wpt/mediarecorder/MediaRecorder-webm-AV-audio-video-dataavailable.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/77119 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/15923 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/75899 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76382 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/20769 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/19190 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/8706 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13973 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/15683 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/20991 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/15424 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/18873 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/17206 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/page/EventHandler.cpp:
```diff
@@ -1586,6 +1587,10 @@ std::optional<Cursor> EventHandler::selectCursor(const HitTestResult& result, bo
     const Cursor& iBeam = horizontalText ? iBeamCursor() : verticalTextCursor();
 
+    // area element has display: none set by default, should use node to get style instead of renderer.
+    if (dynamicDowncast<HTMLAreaElement>(node))
```
We should use `is<HTMLAreaElement>(node)` instead.

AlexsanderDamaceno committed:
cursor property is not applying on <area> element
https://bugs.webkit.org/show_bug.cgi?id=221762

Reviewed by Ryosuke Niwa.

Match with other browsers behavior for area element.

The `<area>` element has `display: none` by default, so no renderer is set for it.
As a result, the cursor could not be looked up in the style object,
since we access the renderer to retrieve cursor information on each mouse event.
This is resolved by accessing the style information directly from the node object when dealing with `<area>` elements.

* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::selectCursor):

Canonical link: https://commits.webkit.org/290169@main
